pkg/controller/v1alpha2/applicationconfiguration/render.go

```diff
@@ -266,6 +272,36 @@ func (r *components) getComponent(ctx context.Context, acc v1alpha2.ApplicationC
 	return c, revisionName, nil
 }
 
+// pass through labels and annotation from app-config to workload  or trait
+func (r *components) passThroughObjMeta(oMeta metav1.ObjectMeta, u *unstructured.Unstructured) {
+	mergeFn := func(src map[string]string, dst *map[string]string) {
```

Comment:
Why use pointer for dst? In fact, variable of map[string]string is already a pointer

Reply:
go maps is not complete references. it only passed by reference once initialized
To avoid panic,  check the dst whether it is initialized or not, if not mergeFn will initialize it
```
package main

import "fmt"

func fn(m map[int]int) {
	m = make(map[int]int)
}

func main() {
	var m map[int]int
	fn(m)
	fmt.Println(m == nil) // got true, hope false
}
```

Reply:
I see, can you help add a unit case that the dst will be nil?

Reply:
this unit covers the scenario where dst is empty, it's has been added
```
	t.Run("workload and trait have no labels and annotation", func(t *testing.T) {
		var u unstructured.Unstructured
		c.passThroughObjMeta(ac.ObjectMeta, &u)
		got := u.GetLabels()
        ...
```

Reply:
Usually we return the result and reduce variables:
```

mergeFn  := func(src map[string]string, dst map[string]string) map[string]string {
}

u.SetLabels(mergeFn(oMeta.GetAnnotations(), u.GetAnnotations()))
```